### PR TITLE
reduce memory bandwitdh to GPU for skinned meshes

### PR DESCRIPTION
**Pull request by AndrewMeadows**
* store skinned mesh cluster weights as normalized uint8
* store skinned mesh cluster indices as uint8 or if necessary uint16

### CODE REVIEW

File: libraries/fbx/src/FBXReader.cpp [hunk 1]
```diff
@@ -1662,61 +1664,69 @@ FBXGeometry* FBXReader::extractFBXGeometry(const QVariantHash& mapping, const QS
...
+                float totalWeight = weightAccumulators[j];
+                for (int k = j + 1; k < j + WEIGHTS_PER_VERTEX; ++k) {
+                    totalWeight += weightAccumulators[k];
+                }
+                if (totalWeight > 0.0f) {
+                    const float ALMOST_HALF = 0.499f;
+                    float weightScalingFactor = (float)(UINT8_MAX) / totalWeight;
+                    for (int k = j; k < j + WEIGHTS_PER_VERTEX; ++k) {
+                        extracted.mesh.clusterWeights[k] = (uint8_t)(weightScalingFactor * weightAccumulators[k] + ALMOST_HALF);
+                    }
```
Comment (AndrewMeadows) [inline libraries/fbx/src/FBXReader.cpp]: The FBXMesh::clusterWeights are being stored as "normalized" uint8's.

File: libraries/fbx/src/FBXReader.h [hunk 2]
```diff
@@ -211,16 +211,14 @@ class FBXMesh {
     QVector<glm::vec3> colors;
     QVector<glm::vec2> texCoords;
     QVector<glm::vec2> texCoords1;
-    QVector<glm::vec4> clusterIndices;
-    QVector<glm::vec4> clusterWeights;
-    
+    QVector<uint16_t> clusterIndices;
+    QVector<uint8_t> clusterWeights;
```
Comment (AndrewMeadows) [inline libraries/fbx/src/FBXReader.h]: Before we know how many we need, we store the `FBXMesh::clusterIndices` as `uint16`.  Later when we generate the gpu mesh we reduce to `uint8` if possible.

**Comment by AndrewMeadows** [comment]
The purpose of this PR is to reduce the memory footprint of the gpu buffer required for each skinned mesh.  In theory this should actually reduce the time spent rendering many avatars.  I would present trace data that proves this but I can't seem to get into the test domains yet.  In any case...

Summary of what to test:

There are no expected changes to behavior in this PR except maybe render speed (this PR should be faster) and gpu memory (this PR should be lower).   The changes specifically relate to how we generate the gpu data for skinned models (e.g. avatars and any "animated" model).  If moving avatars and animated models still work correctly then: good.

BTW, there is an animated robot in dev-playa that could be used to verify this PR.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 92b41d7a3848b01c4b87a0e4b3c3a3a2db935622 available at: https://gist.github.com/18e8a41f3586869ecaac14c3343a4b0b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10026-92b41d7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10026-92b41d7.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5fc1e9716605871eb74b5fe8e1d4e860600c44d4 available at: https://gist.github.com/18e8a41f3586869ecaac14c3343a4b0b<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10026-5fc1e97.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10026-5fc1e97.dmg

**Comment by AndrewMeadows** [comment]
There may have been a slight render speed improvement: about 50 usec less time in `Application::paintGL` (6.664 to 6.615 msec), all of it in `RenderDeferredTasks` (5.267 to 5.205 msec).

**Comment by OmegaHeron** [comment]
Tested Windows 10 - i5 6500 CPU - GTX970 GPU
QA Pass

There are no expected changes to behavior in this PR except maybe render speed (this PR should be faster) and gpu memory (this PR should be lower).  *Yes - minuscule FPS Increase and lowered GPU RAM use*

If moving avatars and animated models still work correctly then: good. *Yes - confirmed via bot in dev-playa and 19 avatars in dev-distributed.  This PR behaves same as latest Master and last RC*

General smoke test passes.

**Pull request merged**